### PR TITLE
csi: remove attacher sidecar from CephFS rook deployment

### PR DESCRIPTION
**Pull request by yati1998**
CephFS CSI driver don't have/advertise controller publish/unpublish
capabilities, thus don't need attacher sidecar for its operations.
The presence of external-attacher adds overhead and issues wrt
attachment in various scenarios. One of them would be the lack of
performance on syncing volumeattachment from API server..etc.
More or less we don't have controller publish and unpublish capabilities,
so we should not make use of this sidecar and cause
unnecessary addon here thus other issues.

similar changes have been added to CSI
https://github.com/ceph/ceph-csi/pull/3149

Signed-off-by: yati1998 <ypadia@redhat.com>

### CODE REVIEW

File: deploy/charts/rook-ceph/templates/clusterrole.yaml [hunk 1]
```diff
@@ -454,12 +454,6 @@ rules:
   - apiGroups: [""]
     resources: ["events"]
     verbs: ["list", "watch", "create", "update", "patch"]
-  - apiGroups: ["storage.k8s.io"]
-    resources: ["volumeattachments"]
-    verbs: ["get", "list", "watch", "patch"]
-  - apiGroups: ["storage.k8s.io"]
-    resources: ["volumeattachments/status"]
-    verbs: ["patch"]
```
Comment (humblec) [inline deploy/charts/rook-ceph/templates/clusterrole.yaml:462]: Removing this global settings could cause issues for RBD deployment @yati1998

**Review by humblec — COMMENTED** [review]
Change /removal from global setting could cause issues for RBD deployment.

File: deploy/charts/rook-ceph/templates/clusterrole.yaml (hunk 1 above)
Comment (Madhu-1) [reply deploy/charts/rook-ceph/templates/clusterrole.yaml:462]: This RBAC is specific to cephfs and will not cause any issue to RBD.

**Review by Madhu-1 — CHANGES_REQUESTED** [review]
* please update all the examples `kubectl get po` outputs (if required as we  have one less container in cephfs deployment)

* update the csidriver object to make attachRequired as false.

File: deploy/charts/rook-ceph/templates/clusterrole.yaml (hunk 1 above)
Comment (humblec) [reply deploy/charts/rook-ceph/templates/clusterrole.yaml:462]: Yeah.. noticed that and was discussing the same with @yati1998 to ignore above.

**Comment by humblec** [comment]
> * please update all the examples `kubectl get po` outputs (if required as we  have one less container in cephfs deployment)
> 
>     * update the csidriver object to make attachRequired as false.

@yati1998 we also need to update the documentation whereever its applicable

**Comment by yati1998** [comment]
> > * please update all the examples `kubectl get po` outputs (if required as we  have one less container in cephfs deployment)
> >   
> >   * update the csidriver object to make attachRequired as false.
> 
> @yati1998 we also need to update the documentation whereever its applicable

Not much to do with the documentation as most of them are global for both cephfs and rbd. Did made few changes anyway.

**Review by humblec — APPROVED** [review]

**Review by travisn — APPROVED** [review]
LGTM, the CI issue looks unrelated to this PR and hopefully stabilized with #10668.

**Comment by humblec** [comment]
@yati1998 please rebase

**Review by Madhu-1 — APPROVED** [review]

**Comment by travisn** [comment]
@Madhu-1 @yati1998 Is this needed to backport to 1.9, or is 1.10 ok? If just 1.10 we could add to PendingReleaseNotes.md about this change. But I'm ok with the backport if needed too.

**Pull request merged**